.travis.yml

```diff
@@ -9,109 +10,90 @@ cache:
+addons:
+  apt:
+    sources:
+      - sourceline: 'ppa:ubuntu-toolchain-r/test'
+    packages:
+      - g++-7
+      - g++-8
+      - g++-9
+      - lcov
 
-linux-gcc-8: &linux-gcc-8
-  os: linux
-  compiler: 'g++-8'
-  addons:
-    apt:
-      sources: ['ubuntu-toolchain-r-test']
-      packages: ['g++-8']
-  before_install:
-    - export CC="gcc-8" CXX="g++-8"
-
-linux-gcc-9: &linux-gcc-9
-  os: linux
-  compiler: 'g++-9'
-  addons:
-    apt:
-      sources: ['ubuntu-toolchain-r-test']
-      packages: ['g++-9']
-  before_install:
-    - export CC="gcc-9" CXX="g++-9"
-
-# https://docs.travis-ci.com/user/languages/c/#gcc-on-linux
-matrix:
+# https://docs.travis-ci.com/user/languages/c
+jobs:
   include:
-  - << : *linux-gcc-7
-    env:
-      - BUILD=coverage
-      - BUILD_TYPE=Debug
-      - CMAKE_VERSION=3.7.2
-  - << : *linux-gcc-9
-    env:
-      - BUILD=unit
-      - BUILD_TYPE=Release
-      - CXXFLAGS="-std=c++2a"
-      - CMAKE_VERSION=3.7.2
-  - << : *linux-gcc-8
-    env:
-      - BUILD=unit
-      - BUILD_TYPE=Release
-      - CMAKE_VERSION=3.7.2
-  - << : *linux-gcc-7
-    env:
-      - BUILD=unit
-      - BUILD_TYPE=Release
-      - CMAKE_VERSION=3.7.2
-  # To reduce build time we disable debug builds of unit tests, because they
-  # are completely contained in coverage test.
-  # - << : *linux-gcc-7
-  #   env:
-  #     - BUILD=unit
-  #     - BUILD_TYPE=Debug
-  - << : *linux-gcc-7
-    env:
-      - BUILD=performance
-      - BUILD_TYPE=Release
-      - CMAKE_VERSION=3.7.2
-  - << : *linux-gcc-7
-    env:
-      - BUILD=header
-      - BUILD_TYPE=Release
-      - CMAKE_VERSION=3.7.2
-      - CXXFLAGS="-Wno-deprecated-declarations"
-  - << : *linux-gcc-7
-    env:
-      - BUILD=snippet
-      - BUILD_TYPE=Release
-      - CMAKE_VERSION=3.7.2
-  - os: linux
-    compiler: 'doxygen'
-    addons:
-      apt:
-        # adds epstopdf, ghostscript, latex
-        packages: ['texlive-font-utils', 'ghostscript', 'texlive-latex-extra']
-    env:
-      - BUILD=documentation
-      - CMAKE_VERSION=3.7.2
-    cache:
-      directories:
+    - name: "Coverage gcc7"
+      env:
+        - CXX=g++-7
+        - CC=gcc-7
+        - BUILD=coverage
+        - BUILD_TYPE=Debug
+    - name: "Unit gcc9 (c++2a)"
+      env:
+        - CXX=g++-9
+        - CC=gcc-9
+        - BUILD=unit
+        - BUILD_TYPE=Release
+        - CXXFLAGS="-std=c++2a"
+    - name: "Unit gcc8"
+      env:
+        - CXX=g++-8
+        - CC=gcc-8
+        - BUILD=unit
+        - BUILD_TYPE=Release
+    - name: "Unit gcc7"
+      env:
+        - CXX=g++-7
+        - CC=gcc-7
+        - BUILD=unit
+        - BUILD_TYPE=Release
+    - name: "Performance gcc7"
+      env:
+        - CXX=g++-7
+        - CC=gcc-7
+        - BUILD=performance
+        - BUILD_TYPE=Release
+    - name: "Header gcc7"
+      env:
+        - CXX=g++-7
+        - CC=gcc-7
+        - BUILD=header
+        - BUILD_TYPE=Release
+        - CXXFLAGS="-Wno-deprecated-declarations"
```

Comment:
Can you try to remove this flag? We shouldn't suppress some warnings globally only because we had some issues at one point in time.

Reply:
Works 👍

Reply:
Nice :) I feared that we hid some errors